controllers/sink.go

```diff
@@ -49,33 +45,33 @@ func (r *SinkReconciler) ObserveSinkStatefulSet(ctx context.Context, req ctrl.Re
 	}, statefulSet)
 	if err != nil {
 		if errors.IsNotFound(err) {
-			r.Log.Info("sink is not ready yet...")
-			return nil
+			r.Log.Info("sink statefulset is not found...")
+		} else {
+			sink.Status.Conditions[v1alpha1.StatefulSet] = condition
+			return err
```

Comment:
if the error is `NotFound`, we should set the status to `create`  and return immediately to let the `apply` phase take actions.

if the error is other reason, we don't know what to do, so not touch the status but return the err is a desired behavior.

Reply:
yeah, for notfound should return early as later code will try to access status and cause NPE.
but for other error, I'm sure if it's good to left status empty ? thought we fail to fetch statefulset, but we can still update with what we observed which is Statefulset Ready=False?